modules/carto/src/layers/raster-tile-layer.ts

```diff
@@ -50,18 +50,16 @@ export default class RasterTileLayer<
     if (!tileJSON) return null;
 
     const {tiles: data, minzoom: minZoom, maxzoom: maxZoom} = tileJSON;
-    return [
-      // @ts-ignore
-      new TileLayer(this.props, {
-        id: `raster-tile-layer-${this.props.id}`,
-        data,
-        // TODO: Tileset2D should be generic over TileIndex type
-        TilesetClass: QuadbinTileset2D as any,
-        renderSubLayers,
-        minZoom,
-        maxZoom,
-        loadOptions: this.getLoadOptions()
-      })
-    ];
+    const SubLayerClass = this.getSubLayerClass('tile', TileLayer);
+    return new SubLayerClass(this.props, {
+      id: `raster-tile-layer-${this.props.id}`,
+      data,
+      // TODO: Tileset2D should be generic over TileIndex type
+      TilesetClass: QuadbinTileset2D as any,
+      renderSubLayers,
```

Comment:
I was thinking, that if we allow overriding `TileLayer` class, why not allowing overriding of `RasterLayer` too?

(that was something i wanted to do when experimenting with this class)

Reply:
I did consider that, but we have no precedent doing that elsewhere in `TileLayer` or subclasses. The design of `TileLayer` is to provide the `renderSubLayers` prop to allow the application to set (or override) the correct class. In this layer we hardcode it, preventing it from being set from the outside. Perhaps it would be better to move the definition of `renderSubLayers` to defaultProps like `TileLayer` does it?